cmd/run.go

```diff
@@ -572,3 +567,62 @@ func logVersion() {
 		"os/arch", fmt.Sprintf("%s/%s", runtime.GOOS, runtime.GOARCH),
 	)
 }
+
+func forkIDIntervals(ctx context.Context, st *state.State, etherman *etherman.Client, genesisBlockNumber uint64) ([]state.ForkIDInterval, error) {
+	dbTx, err := st.BeginStateTransaction(ctx)
```

Comment:
I think BeginStateTransaction can be placed just before the for loop to store forkids in the db (line 598) and do the commit just after end the loop (line 605). Anyway we are not doing a rollback when returning from this function with an error. If we move the BeginStateTransaction just before the for loop, then we need only to do a rollback if we have an error when inserting a forkid in the db